cognite/client/data_classes/data_modeling/queries.py

```diff
@@ -0,0 +1,343 @@
+            edges["maxDistance" if camel_case else "max_distance"] = self.max_distance
+        if self.direction:
+            edges["direction"] = self.direction
+        if self.filter:
+            edges["filter"] = self.filter.dump()
+        if self.node_filter:
+            edges["nodeFilter" if camel_case else "node_filter"] = self.node_filter.dump()
+        if self.termination_filter:
+            edges["terminationFilter" if camel_case else "termination_filter"] = self.termination_filter.dump()
+        if self.limit_each:
+            edges["limitEach" if camel_case else "limit_each"] = self.limit_each
+
+        if self.sort:
+            output["sort"] = [s.dump(camel_case=camel_case) for s in self.sort]
+        if self.post_sort:
+            output["postSort" if camel_case else "post_sort"] = [s.dump(camel_case=camel_case) for s in self.post_sort]
+        if self.limit:
+            output["limit"] = self.limit
+
+        return output
+
+
+class QueryResult(UserDict):
+    def __getitem__(self, item: str) -> NodeList | EdgeList:
+        return super().__getitem__(item)
+
+    @property
+    def cursors(self) -> dict[str, str]:
+        return {key: value.cursor for key, value in self.items()}
+
+    @classmethod
+    def load(
+        cls, data: dict[str, Any] | str, default_by_reference: dict[str, Type[NodeList] | Type[EdgeList]]
+    ) -> QueryResult:
+        data = json.loads(data) if isinstance(data, str) else data
+        instance = cls()
+        for key, values in data.items():
+            if not values:
+                instance[key] = default_by_reference[key]([])
+            elif values[0].get("instanceType") == "node":
+                instance[key] = NodeList._load(values)
+            elif values[0].get("instanceType") == "edge":
+                instance[key] = EdgeList._load(values)
+            else:
+                raise ValueError(f"Unexpected instance type {values[0].get('instanceType')}")
+        return instance
+
+    def update_json(
+        self, data: dict[str, Any] | str, default_by_reference: dict[str, Type[NodeList] | Type[EdgeList]]
+    ) -> None:
+        data = json.loads(data) if isinstance(data, str) else data
+        for key, values in data.items():
+            if key not in self:
+                self[key] = default_by_reference[key]([])
+            if isinstance(self[key], NodeList):
+                self[key].extend([Node.load(v) for v in values])
+            elif isinstance(self[key], EdgeList):
+                self[key].extend([Edge.load(v) for v in values])
+            else:
+                raise ValueError(f"Unexpected instance type {type(self[key])}")
+
+    def update_cursors(self, cursors: dict[str, Any] | None) -> None:
+        cursors = cursors or {}
+        for key, value in self.items():
+            # The sync endpoint eventually returns None, so for that case we keep the last cursor.
+            value.cursor = cursors.get(key, value.cursor)
```

Comment:
Let's get rid of these. We can deserialize within `load` without needing to mutate the object after.

Reply:
update_json is actually unused already.